src/main/java/com/mineboundteam/minebound/magic/MagicEvents.java

```diff
@@ -117,7 +117,11 @@ public static void onButtonPress(InputEvent.KeyInputEvent event) {
                 MagicSync.NET_CHANNEL.sendToServer(new ButtonMsg(MsgType.SECONDARY_RELEASED));
         }
     }
-
+    @SubscribeEvent
+    public static void onInputEvent(InputEvent event){
+        if(ClientRegistry.FIRE_UTILITY_SPELL_TOGGLE.consumeClick())
+            MagicSync.NET_CHANNEL.sendToServer(new ButtonMsg(MsgType.FIRE_UTILITY_TOGGLE));
+    }
```

Comment:
We already have an `InputEvent` function defined above, line 82. You should just stick this `if` in there; now I'm not sure how `InputEvent` handles the player clicking multiple buttons at the same time, but just to play it safe, don't `else if` it with the existing `if` statements since that might prevent the packet from being sent when both the primary/secondary spell select keybind and this keybind is being pressed.

Reply:
This function was made at the request of Mason. It handles both the keyboard and mouse actions without needing to differentiate.

Reply:
Right, we've already subscribed to that event on line 82. No need to make a duplicate event subscriber for it.

Reply:
Ah my bad you're right